util/encoding/encoding.go

```diff
@@ -493,10 +508,16 @@ func DecodeIfNull(b []byte) ([]byte, bool) {
 // empty string encoding, so do not use this routine where it is necessary to
 // distinguish not-NULL from the empty string.
 func DecodeIfNotNull(b []byte) ([]byte, bool) {
-	r := bytes.TrimPrefix(b, encodedNotNull)
-	return r, len(r) < len(b)
+	if len(b) >= 1 && b[0] == encodedNotNull {
```

Comment:
Done on the `PeekType` usage. 

Plumbing through error handling everywhere these methods are used will be irritating. And I don't think I've ever seen these panics fire. I have seen other panics from the SQL code (e.g. index out of bounds). We need to make sure we have a appropriate recovery in those situations and return some sort of internal server error.